lib/board.rb

```diff
@@ -0,0 +1,130 @@
+class Board
+  attr_reader(:current_player, :board_middle, :turn_count, :max_turns, :board_size)
+  BOARD_ROW = '-'
+  BOARD_COLUMM = '|'
+  BEGINNING_AND_END_LENGTH = 2
+  ITERATE_RISING = -1
+  ITERATE_FALLING = 1
+  ITERATE_FROM_START = 0
+
+  def initialize(board_size = 3, command_line_application = true)
+    @board = []
+    @board_middle = []
+    @board_size = board_size
+    @max_turns = board_size * board_size
+    set_board(command_line_application)
+  end
+
+  def set_board(command_line_application)
+    set_middle_sections
+    view_board if command_line_application
+  end
+
+  def set_middle_sections
+    middle = * (1..@max_turns)
+    middle = middle.map(&:to_s)
+    @board_middle = middle.each_slice(@board_size).to_a
+  end
+
+  def view_board
+    system('clear')
+    reset_board
+    assemble_board
+    puts @board
+  end
+
+  def check_square_is_free(row, column)
+    @board_middle[row][column] != 'X' && @board_middle[row][column] != 'O'
+  end
+
+  def make_move(current_player, row, column)
+    @board_middle[row][column] = current_player if check_square_is_free(row, column)
+  end
+
+  def check_for_winner(current_player)
+    check_row || check_column(current_player) ||
+      check_diagonal(current_player, ITERATE_FROM_START, ITERATE_FALLING) ||
+      check_diagonal(current_player, @board_size - 1, ITERATE_RISING)
+  end
+
+  private
+
+  def board_top_and_bottom
+    board_top_and_bottom = []
+    (@board_size + BEGINNING_AND_END_LENGTH).times do
+      board_top_and_bottom.push(BOARD_ROW)
+    end
+    board_top_and_bottom = [board_top_and_bottom.join]
+  end
+
+  def assemble_board
```

Comment:
Assembling the board to be displayed feels like it’s not responsibility of the board itself. I would extract those functions into a different class